packages/aws-cdk-lib/custom-resource-config/test/custom-resource-config.test.ts

```diff
@@ -0,0 +1,547 @@
+        });
+        const destination = new events.ApiDestination(stack, "Destination", {
+            connection,
+            endpoint: "https://amazon.com",
+            description: "Calling example.com with API key x-api-key",
+        });
+        const rule = new events.Rule(stack, "Rule", {
+            schedule: events.Schedule.rate(cdk.Duration.minutes(1)),
+            targets: [new targets.ApiDestination(destination)],
+        });
+        const awsApi = new targets.AwsApi({
+            action: "action",
+            service: "service",
+            apiVersion: "apiVersion",
+            catchErrorPattern: "catchErrorPattern",
+            parameters: "hello-world",
+            policyStatement: new iam.PolicyStatement({
+                actions: ["s3:GetObject"],
+                resources: ["*"],
+            }),
+        });
+        rule.addTarget(awsApi);
+
+        const template = Template.fromStack(stack);
+
+        template.resourceCountIs('AWS::Logs::LogGroup', 1);
+        template.hasResourceProperties('AWS::Logs::LogGroup', {
+            RetentionInDays: SET_LOG_RETENTION
+        })
+        template.hasResourceProperties('AWS::Lambda::Function', {
+            LoggingConfig: {
+                LogGroup: {
+                    Ref: Match.stringLikeRegexp('AWS')
+                }
+            }
+        })
+    })
+    test('awsapi - snapshot', () => {
+        const SET_LOG_RETENTION = logs.RetentionDays.TEN_YEARS;
+
+        const app = new cdk.App();
+        CustomResourceConfig.of(app).addLogRetentionLifetime(SET_LOG_RETENTION);
+        const stack = new cdk.Stack(app);
+
+        const connection = new events.Connection(stack, "Connection", {
+            authorization: events.Authorization.apiKey(
+                "x-api-key",
+                cdk.SecretValue.unsafePlainText("no-secret")
+            ),
+            description: "Connection with API Key x-api-key",
+        });
+        const destination = new events.ApiDestination(stack, "Destination", {
+            connection,
+            endpoint: "https://amazon.com",
+            description: "Calling example.com with API key x-api-key",
+        });
+        const rule = new events.Rule(stack, "Rule", {
+            schedule: events.Schedule.rate(cdk.Duration.minutes(1)),
+            targets: [new targets.ApiDestination(destination)],
+        });
+        const awsApi = new targets.AwsApi({
+            action: "action",
+            service: "service",
+            apiVersion: "apiVersion",
+            catchErrorPattern: "catchErrorPattern",
+            parameters: "hello-world",
+            policyStatement: new iam.PolicyStatement({
+                actions: ["s3:GetObject"],
+                resources: ["*"],
+            }),
+        });
+        rule.addTarget(awsApi);
+
+        const template = Template.fromStack(stack);
+
+        expect(template.toJSON()).toMatchSnapshot();
+    })
+})
+
+describe('LogGroup', () => {
```

Comment:
`LogGroup` does not provide enough information. Are we modifying existing Log Groups? Deleting them? Creating them? Something else entirely?